sdk/deepbook-v3/src/client.ts

```diff
@@ -567,4 +567,33 @@ export class DeepBookClient {
 			},
 		};
 	}
+
+	/**
+	 * @description Get the vault balances for a pool
+	 * @param {string} poolKey Key of the pool
+	 * @returns {Promise<{ base: number, quote: number, deep: number }>}
+	 * An object with base, quote, and deep balances in the vault
+	 */
+	async lockedBalance(poolKey: string, balanceManagerKey: string) {
+		const tx = new Transaction();
+		const pool = this.#config.getPool(poolKey);
+		const baseScalar = this.#config.getCoin(pool.baseCoin).scalar;
+		const quoteScalar = this.#config.getCoin(pool.quoteCoin).scalar;
+
+		tx.add(this.deepBook.lockedBalance(poolKey, balanceManagerKey));
+		const res = await this.client.devInspectTransactionBlock({
+			sender: normalizeSuiAddress(this.#address),
+			transactionBlock: tx,
+		});
+
+		const baseLocked = Number(bcs.U64.parse(new Uint8Array(res.results![0].returnValues![0][0])));
+		const quoteLocked = Number(bcs.U64.parse(new Uint8Array(res.results![0].returnValues![1][0])));
```

Comment:
These are all unsafe number conversions fwiw, should be using bigint

Reply:
Noted, thank you

Reply:
Will address in an improvement PR outside this one, for now rounding to 9 decimals

Reply:
You'll actually have the precision issue before you do the division because JS can only represent numbers up to 2^53 - 1, but fine to punt.